Yank/multistate/replicaexchange.py

```diff
@@ -352,6 +352,8 @@ def _mix_neighboring_replicas(self):
         """Attempt exchanges between neighboring replicas only."""
         logger.debug("Will attempt to swap only neighboring replicas.")
 
+        # TODO: Extend this to allow more remote swaps or more thorough mixing if locality > 1.
```

Comment:
I really think we can easily adapt the existing Cython code since you have to feed it all the data manually, so we can just slice out the `states` we can swap to, and run it through the code

Reply:
Extending the `all-swaps` cython code is certainly nontrivial. At the very least, it would require explicitly *rejecting* proposed moves that propose an exchange where one of the replicas has no energy evaluated at a new state, which means we have to also feed in the neighborhoods matrix.

We can also extend the `neighbor-swap` scheme to allow more remote proposals if `locality > 1`, but the acceptance probability falls off more rapidly.

I think these are great things to explore, but probably in a separate PR.

Reply:
> I think these are great things to explore, but probably in a separate PR.

great idea!